go.mod: move golang.org/x/tools to master + fix knz/go-libedit version

otan opened the pull request:
* Repoint golang.org/x/tools to master as it now contains the goyacc
  change
* Point knz/go-libedit back at a valid location (this still gets
  replaced), as I noticed people have difficulty when pulling sometimes
  when it points at an invalid v0.0.0 version. This has no effect on
  vendor in practice.

Release note: None

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.png" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/58425)
<!-- Reviewable:end -->

otan commented:
looks like the agent pool makes all my PRs red 😬

otan committed:
go.mod: move golang.org/x/tools to master + fix knz/go-libedit version

* Repoint golang.org/x/tools to master as it now contains the goyacc
  change
* Point knz/go-libedit back at a valid location (this still gets
  replaced), as I noticed people have difficulty when pulling sometimes
  when it points at an invalid v0.0.0 version. This has no effect on
  vendor in practice.

Release note: None

otan commented:
bors r=irfansharif
